sale_procurement_group_by_line/model/sale.py

```diff
@@ -28,143 +14,33 @@ class SaleOrder(models.Model):
-                    proc_ids += [x.id for x in line.procurement_ids
-                                 if x.state in ('exception', 'cancel')]
-                elif sale_line_obj.need_procurement(cr, uid, [line.id],
-                                                    context=context):
-                    if (line.state == 'done') or not line.product_id:
-                        continue
-                    vals = self._prepare_order_line_procurement(
-                        cr, uid, order, line,
-                        group_id=group_id, context=context)
-                    proc_id = procurement_obj.create(
-                        cr, uid, vals, context=context)
-                    proc_ids.append(proc_id)
-            # Confirm procurement order such that rules will be applied on it
-            # note that the workflow normally ensure proc_ids isn't an empty
-            # list
-            procurement_obj.run(cr, uid, proc_ids, context=context)
-
-            # if shipping was in exception and the user choose to recreate the
-            # delivery order, write the new status of SO
-            if order.state == 'shipping_except':
-                val = {'state': 'progress', 'shipped': False}
-
-                if (order.order_policy == 'manual'):
-                    for line in order.order_line:
-                        if (not line.invoiced and
-                                line.state not in ('cancel', 'draft')):
-                            val['state'] = 'manual'
-                            break
-                order.write(val)
-        return True
-
-    ###
-    # OVERRIDE to use sale.order.line's procurement_group_id from lines
-    ###
-    @api.one
-    @api.depends('order_line.procurement_group_id.procurement_ids.state')
-    def _get_shipped(self):
-        """ As procurement is per sale line basis, we check each line
-
-            If at least a line has no procurement group defined, it means it
-            isn't shipped yet.
-
-            Only when all procurement are done or cancelled, we consider
-            the sale order as being shipped.
-
-            And if there is no line, we have nothing to ship, thus it isn't
-            shipped.
-
-        """
-        if not self.order_line:
-            self.shipped = False
-            return
-
-        # keep empty groups
-        groups = set([line.procurement_group_id
-                      for line in self.order_line
-                      if line.product_id.type != 'service'])
-        is_shipped = True
-        for group in groups:
-            if not group or not group.procurement_ids:
-                is_shipped = False
-                break
-            is_shipped &= all([proc.state in ['cancel', 'done']
-                               for proc in group.procurement_ids])
-        self.shipped = is_shipped
+        return self._prepare_procurement_group()
 
     ###
     # OVERRIDE to find sale.order.line's picking
     ###
```

Comment:
I know he refers to that method:
Is it possible to override this method to calculate the pickings related to a SO line and keep the current functionality? I know how to update two different computed fields in a single method for other field types but not for many2many.

If it's possible then keep it but if it's not I think this can be deleted IMHO.

Regards